test/runtests.jl

```diff
@@ -1,39 +1,57 @@
-using ForwardDiff
-
-println("Testing Partials...")
-t = @elapsed include("PartialsTest.jl")
-println("done (took $t seconds).")
-
-println("Testing Dual...")
-t = @elapsed include("DualTest.jl")
-println("done (took $t seconds).")
-
-println("Testing derivative functionality...")
-t = @elapsed include("DerivativeTest.jl")
-println("done (took $t seconds).")
-
-println("Testing gradient functionality...")
-t = @elapsed include("GradientTest.jl")
-println("done (took $t seconds).")
-
-println("Testing jacobian functionality...")
-t = @elapsed include("JacobianTest.jl")
-println("done (took $t seconds).")
-
-println("Testing hessian functionality...")
-t = @elapsed include("HessianTest.jl")
-println("done (took $t seconds).")
-
-println("Testing perturbation confusion functionality...")
-t = @elapsed include("ConfusionTest.jl")
-println("done (took $t seconds).")
-
-println("Testing miscellaneous functionality...")
-t = @elapsed include("MiscTest.jl")
-println("done (took $t seconds).")
-
-if VERSION >= v"1.5-"
-    println("Testing allocations...")
-    t = @elapsed include("AllocationsTest.jl")
-    println("done (took $t seconds).")
-end
+using ForwardDiff, Test, Random
+
+SEED = trunc(Int, time())
+println("##### Random.seed!($SEED), on VERSION == $VERSION")
+Random.seed!(SEED)
```

Comment:
Previously it tested with a random global seed.

What was confusing is that sometimes this gives PRIMAL == PRIMAL2 (for integers), and sometimes not. It would be clearer to deliberately test both cases. But, for now, I just made it print the seed used, so that you can lock it to debug locally. 

Fixing the seed completely would test only one case per version of Julia.